### PR TITLE
extend resetOpenNodes

### PR DESCRIPTION
**Pull request by tagyoureit**
This is a solution for #109.  If the openNodes function is extended to accept a key as well as nodes path it would be exactly what I need.

This function could be called on the base <TreeMenu ... /> component without any modifications at all to the rest of the code.  See [Stackblitz](https://stackblitz.com/edit/react-ts-gswhar) for an example of this using the existing functionality of openNodes.  This PR would also highlight the activeKey as well.  Usage example with the PR would be `this.childRef.current.resetOpenNodes(['mammal'], 'mammal')`.  Thanks!

### CODE REVIEW

File: src/TreeMenu/index.tsx [hunk 1]
```diff
@@ -63,11 +63,11 @@ class TreeMenu extends React.Component<TreeMenuProps, TreeMenuState> {
     }
   }
 
-  resetOpenNodes = (newOpenNodes?: string[]) => {
+  resetOpenNodes = (newOpenNodes?: string[], key?: string) => {
```
Comment (iannbing) [inline src/TreeMenu/index.tsx]: Would it be slightly better also have `activeKey` and `focusKey` instead of `key`?

File: src/TreeMenu/index.tsx [hunk 2]
```diff
@@ -63,11 +63,11 @@ class TreeMenu extends React.Component<TreeMenuProps, TreeMenuState> {
...
     const { initialOpenNodes } = this.props;
     const openNodes =
       (Array.isArray(newOpenNodes) && newOpenNodes) || initialOpenNodes || [];
-    this.setState({ openNodes, searchTerm: '' });
+    this.setState({ openNodes, searchTerm: '', activeKey: key || '', focusKey: '' });
```
Comment (iannbing) [inline src/TreeMenu/index.tsx]: Nice that you also take into account `focusKey`. Why not also allow modifying `focusKey`? 🙂

**Comment by tagyoureit** [comment]
Done, done and done... and I added a Storybrook demo as well.

**Comment by tagyoureit** [comment]
Did you want to increase the version in NPM/package.json?  Otherwise it won't show as a new version?  thx for including this.

**Comment by iannbing** [comment]
I've released version `1.1.12`. Let me know if you have further issues :-).

**Pull request merged**